test/hga/HGA_Regression.ipynb

```diff
@@ -1474,47 +1468,25 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "\"\"\"\n",
+    "import re\n",
+    "\n",
     "if environment_information is not None:\n",
     "    granules = {Environment.PROD: 'G1715962900-ASF',\n",
     "                Environment.UAT: 'G1244144618-EEDTEST',\n",
     "                Environment.SIT: 'G1244144618-EEDTEST'}\n",
     "\n",
-    "    sentinel_directory_thirteen = 'sentinel_test_thirteen'\n",
-    "\n",
     "    sentinel_request_thirteen = Request(collection=environment_information['sentinel_collection'],\n",
     "                                        spatial=BBox(-115.45, 32.95, -115.15, 33.09),\n",
     "                                        granule_id=granules[environment_information['env']])\n",
     "\n",
-    "    sentinel_output_thirteen = make_request_and_download_result(harmony_client, sentinel_request_thirteen,\n",
-    "                                                                sentinel_directory_thirteen)\n",
-    "\n",
-    "    sentinel_variables_thirteen = {'amplitude',\n",
-    "                                   'coherence',\n",
-    "                                   'connectedComponents',\n",
-    "                                   'unwrappedPhase',\n",
-    "                                   'lat',\n",
-    "                                   'lon',\n",
-    "                                   'latitude_longitude'}\n",
-    "\n",
-    "    expected_sentinel_results_thirteen = {'authority': None,\n",
-    "                                          'cs': None,\n",
-    "                                         'gcs': None,\n",
-    "                                         'gcs_epsg': None,\n",
-    "                                         'n_bands': 7,\n",
-    "                                         'proj_cs': None,\n",
-    "                                         'proj_epsg': None,\n",
-    "                                         'reference_image': 'reference_images/sentinel_reference_seven.nc',  # Should be same output as test case 7.\n",
-    "                                         'spatial_extent': [32.95, 33.09, -115.45, -115.15],\n",
-    "                                         'variables': sentinel_variables_thirteen,\n",
-    "                                         'width': 360,\n",
-    "                                         'height': 168}\n",
-    "    \n",
-    "    check_request_output(sentinel_output_thirteen[0], expected_sentinel_results_thirteen)\n",
-    "    clean_test_artefacts(sentinel_directory_thirteen)\n",
+    "    sentinel_job_id_thirteen = harmony_client.submit(sentinel_request_thirteen)\n",
+    "    try: \n",
+    "        harmony_client.wait_for_processing(sentinel_job_id_thirteen, show_progress=True)\n",
+    "    except Exception as exception:\n",
+    "        assert re.search(\"Request cannot be completed: datasets are incompatible and cannot be combined.\", str(exception)), 'Exception not raised correctly'\n",
```

Comment:
Nitpick: `re.search` is a pretty computationally expensive way to search for a substring, maybe you just use:

```
assert 'Request cannot be completed: dataset are incompatible and cannot be combined.' in str(exception)
```

Reply:
I'd argue we don't change it until we see we need to by profiling. But you can win on readability and pythonic code right now. I can fix this. 5785404